File: lang/go/test/vac.fun/data_structures/LinkedStack/LinkedStackTest.go
package main

import (
    "vac.fun/data_structures/LinkedStack"
    "vac.fun/util/TestRunner"
)

func testLinkedStack() bool {
    size := 8192

    stack := LinkedStack.New()

    for i := 1; i <= size; i++ {
        stack.Push(i)
    }

    if stack.Size() != size {
        return false
    }

    for i := size; i > 0; i-- {
        if stack.Peek() != i {
            return false
        }
        stack.Pop()
    }

    return stack.IsEmpty()
}

func main() {
    TestRunner.ParseTest(testLinkedStack())
}

File: python/muse/data_structures/LinkedStackTest.py
from muse.data_structures.LinkedStack import LinkedStack
from muse.util import TestRunner

def testLinkedStack() -> bool:
    size = 8192

    stack = LinkedStack()

    for i in range(1, size + 1):
        stack.push(i)

    if stack.size() != size:
        return False

    for i in range(size, 0, -1):
        if stack.peek() != i:
            return False
        stack.pop()

    return stack.isEmpty()

def main() -> None:
    TestRunner.parseTest(testLinkedStack())

if __name__ == "__main__":
    main()
